lib/clean-host.js

```diff
@@ -89,30 +76,45 @@ module.exports = function extractHostname(value) {
     url = '' + url;
   }
 
-  var needsTrimming = checkTrimmingNeeded(url);
-  if (needsTrimming) {
-    url = url.trim();
-  }
-
   var needsLowerCase = checkLowerCaseNeeded(url);
   if (needsLowerCase) {
     url = url.toLowerCase();
   }
```

Comment:
This diff is a little funky, but I just moved the lowercase check to before the ipv6 check since the later IP checks expect a lowercased string.